keps/2937-resource-transformer/README.md

```diff
@@ -0,0 +1,560 @@
+```yaml
+  nvidia.com/mig-1g.5gb: 2
+  nvidia.com/mig-2g.10gb: 1
+  memory: 100M
+  cpu: 1
+```
+would yield a Workload with an effective request of:
+```yaml
+  example.com/accelerator-memory: 20G
+  memory: 100M
+  cpu: 1
+```
+
+To compare, to achieve the identical effective quotas without `replace` the covered
+resource stanza of each ClusterQueue would need to be:
+```yaml
+...
+  - coveredResources: ["cpu", "memory", "example.com/accelerator-memory", "nvidia.com/mig-1g.5gb", "nvidia.com/mig-2g.10gb", "nvidia.com/mig-4g.20gb", "nvidia.com/mig-7g.40gb"]
+    flavors:
+    - name: default-flavor
+      resources:
+      - name: "cpu"
+        nominalQuota: 8
+      - name: "memory"
+        nominalQuota: 500G
+      - name: example.com/accelerator-memory
+        nominalQuota: 200G
+      - name: nvidia.com/mig-1g.5gb
+        nominalQuota: 40  # 200G/5g
+      - name: nvidia.com/mig-2g.10gb
+        nominalQuota: 20  # 200G/10g
+      - name: nvidia.com/mig-4g.20gb
+        nominalQuota: 10  # 200G/20g
+      - name: nvidia.com/mig-7g.40gb
+        nominalQuota: 5  # 200G/40g
+```
+
+##### Story 1B - Using Retain with MIG
+
+However, some cluster admins may desire to use `retain` for MIG resources to exercise
+finer grained control over the use of MIG partitions.
+To do this, the cluster admin could register the following transformations:
+* (`nvidia.com/mig-1g.5gb`, `example.com/accelerator-memory`, `5Gi`, `retain`)
+* (`nvidia.com/mig-2g.10gb`, `example.com/accelerator-memory`, `10Gi`, `retain`)
+* (`nvidia.com/mig-4g.20gb`, `example.com/accelerator-memory`, `20Gi`, `retain`)
+* (`nvidia.com/mig-7g.40gb`, `example.com/accelerator-memory`, `40Gi`, `retain`)
+
+A Job requesting the resources below:
+```yaml
+  nvidia.com/mig-1g.5gb: 2
+  nvidia.com/mig-2g.10gb: 1
+  memory: 100M
+  cpu: 1
+```
+would yield a Workload with an effective resource request of:
+```yaml
+  example.com/accelerator-memory: 20G
+  nvidia.com/mig-1g.5gb: 2
+  nvidia.com/mig-2g.10gb: 1
+  memory: 100M
+  cpu: 1
+```
+
+Below is an example ResourceGroup that the cluster admin could define that
+prevents the usage of `7g.40gb` paritions and puts a tighter cap on `4g.20gb`
+partitions than would be implied by the `accelerator-memory` quota:
+```yaml
+  - coveredResources: ["cpu", "memory", "example.com/accelerator-memory"]
+    flavors:
+    - name: default-flavor
+      resources:
+      - name: "cpu"
+        nominalQuota: 8
+      - name: "memory"
+        nominalQuota: 500G
+      - name: example.com/accelerator-memory
+        nominalQuota: 200G
+      - name: nvidia.com/mig-1g.5gb
+        nominalQuota: 40  # 200G/5g
+      - name: nvidia.com/mig-2g.10gb
+        nominalQuota: 20  # 200G/10g
+      - name: nvidia.com/mig-4g.20gb
+        nominalQuota: 4  # less than the 10 that would be allowed by the acelerator-memory quota
+      - name: nvidia.com/mig-7g.40gb
+        nominalQuota: 0  # not allowed -- inefficient use of GPU resources
```

Comment:
What happens if we specify the borrowingLimit like this?

```yaml
      - name: nvidia.com/mig-7g.40gb
        nominalQuota: 0
        borrowingLimit: 10
```

Is this borrowingLimit still effective the same way it is today?

Reply:
yes.  no change to borrowingLimit or lendingLimit.  I added a `borrowingLimit: 0`  to the example to clarify.